### PR TITLE
fix some issues for samples repo

### PR DESCRIPTION
**Pull request by frank-dong-ms**
1. upgrade SciSharp.TensorFlow.Redist version to 2.3.1 same as ml.net
2. fix tf not running on netcore 2.1 exception by introducing reference to "System.Runtime.CompilerServices.Unsafe" package

### CODE REVIEW

**Review by harishsk — APPROVED** [review]

**Comment by frank-dong-ms** [comment]
Merge this PR until we upgrade the ML.NET to 1.5.3 used in samples repro as Antonio pointed out.

File: samples/csharp/end-to-end-apps/DeepLearning_ImageClassification_TF/TFImageClassification/TFImageClassification.csproj [hunk 1]
```diff
@@ -11,7 +11,8 @@
     <PackageReference Include="Microsoft.ML" Version="$(MicrosoftMLVersion)" />
     <PackageReference Include="Microsoft.ML.ImageAnalytics" Version="$(MicrosoftMLVersion)" />
     <PackageReference Include="Microsoft.ML.TensorFlow" Version="$(MicrosoftMLVersion)" />
-    <PackageReference Include="SciSharp.TensorFlow.Redist" Version="2.3.0" />
+    <PackageReference Include="SciSharp.TensorFlow.Redist" Version="2.3.1" />
+    <PackageReference Condition="'$(TargetFramework)'=='netcoreapp2.1'" Include="System.Runtime.CompilerServices.Unsafe" Version="4.5.3" />
```
Comment (eerhardt) [inline samples/csharp/end-to-end-apps/DeepLearning_ImageClassification_TF/TFImageClassification/TFImageClassification.csproj:15]: This is kind of odd. Can you explain what the problem is here? In general, this smells like a packaging issue...

Comment (frank-dong-ms) [reply samples/csharp/end-to-end-apps/DeepLearning_ImageClassification_TF/TFImageClassification/TFImageClassification.csproj:15]: Yes, when we try to run TF related samples in netcore 2.1 in recent build (with upgrade to use tensorflow 2 from this PR: [https://github.com/dotnet/machinelearning/pull/5404](https://github.com/dotnet/machinelearning/pull/5404)) we will see below exception so I added the missing reference and that fix the exception:
System.IO.FileLoadException: "Could not load file or assembly "System.Runtime.CompilerServices.Unsafe, Version=4.0.4.1, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" or one of it's dependences

One interesting thing is this exception only happens on netcore2.1 but not netcore 3.1.


---
In reply to: [540966485](https://github.com/dotnet/machinelearning-samples/pull/870#discussion_r540966485) [](ancestors = 540966485)

Comment (frank-dong-ms) [reply samples/csharp/end-to-end-apps/DeepLearning_ImageClassification_TF/TFImageClassification/TFImageClassification.csproj:15]: My guess is package "SciSharp.TensorFlow.Redist" take dependency on "System.Runtime.CompilerServices.Unsafe" with version 4.5.3 and package "System.Runtime.CompilerServices.Unsafe" is not loaded by default on netcore2.1? We will really appreciate if you can help us understand the root cause of this issue.

---
In reply to: [541388519](https://github.com/dotnet/machinelearning-samples/pull/870#discussion_r541388519) [](ancestors = 541388519,540966485)

Comment (eerhardt) [reply samples/csharp/end-to-end-apps/DeepLearning_ImageClassification_TF/TFImageClassification/TFImageClassification.csproj:15]: Can you send me an app that reproduces the failure? I would need to debug it to figure out the problem.

Comment (frank-dong-ms) [reply samples/csharp/end-to-end-apps/DeepLearning_ImageClassification_TF/TFImageClassification/TFImageClassification.csproj:15]: Thanks Eric, use below ml.net sample with netcore2.1 and take "1.5.3-preview.20611.3" version of ml.net as dependency you should see the error:

https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/DeepLearning_ImageClassification_TensorFlow

Comment (frank-dong-ms) [reply samples/csharp/end-to-end-apps/DeepLearning_ImageClassification_TF/TFImageClassification/TFImageClassification.csproj:15]: @eric.erhardt@microsoft.com I will merge this PR first to unblock others. If you can figure out the root cause and have better fix we can discuss later, thanks.

---
In reply to: [541452261](https://github.com/dotnet/machinelearning-samples/pull/870#discussion_r541452261) [](ancestors = 541452261)

Comment (eerhardt) [reply samples/csharp/end-to-end-apps/DeepLearning_ImageClassification_TF/TFImageClassification/TFImageClassification.csproj:15]: Logged https://github.com/dotnet/runtime/issues/46082 for this problem.

**Pull request merged**